iv-outlier-detection/src/adapter.ts

```diff
@@ -0,0 +1,40 @@
+import { Execute } from '@chainlink/types'
+import { Requester, Validator } from '@chainlink/external-adapter'
+import { fetchGenesisVolatility } from './genesisVolatility'
+import { fetchDerbit } from './derbit'
+
+export type ExternalFetch = (symbol: string, days: number) => Promise<number>
+
+const difference = (a: number, b: number): number => {
+  return (Math.abs(a - b) / ((a + b) / 2)) * 100
+}
+
+const customParams = {
+  symbol: ['base', 'from', 'coin', 'symbol'],
+  days: ['days', 'period'],
+}
+
+export const execute: Execute = async (input) => {
+  const validator = new Validator(input, customParams)
+  if (validator.error) throw validator.error
+  if (!input.meta || !input.meta.latestAnswer) throw new Error('missing meta data')
```

Comment:
Where did the `meta` come from? What's the difference vs. `data`, when was it introduced and what do we use it for?

Reply:
The meta data (https://github.com/smartcontractkit/chainlink/pull/3376) provides some information about the on-chain reference contract that we can use, instead of having to specify an RPC_URL to poll (and also specify the contract address in the job spec).

In our case, we're interested in getting the latest answer from the reference contract - which we can now get from the meta data.

I'll add a short entry to the changelog about this.

Reply:
Awesome! Looks super useful to have that extra metadata around.

I missed that, unfortunately. Need to keep an eye on that Chainlink CHANGELOG.